consul/state/prepared_query.go

```diff
@@ -16,22 +17,54 @@ func isUUID(str string) bool {
 	return validUUID.MatchString(str)
 }
 
+// queryWrapper is an internal structure that is used to store a query alongside
+// its compiled template, which can be nil.
+type queryWrapper struct {
+	*structs.PreparedQuery
```

Comment:
Do you ever need to call the methods on a queryWrapper? Trying to understand why the prepared query itself is embedded.